internal/security/credentials.go

```diff
@@ -47,30 +53,81 @@ func (s *SecretProvider) GetDatabaseCredentials(database db.DatabaseInfo) (commo
 // specified path will be returned.
 func (s *SecretProvider) GetSecrets(path string, keys ...string) (map[string]string, error) {
 
+	if cachedSecrets := s.getCachedSecrets(path, keys...); cachedSecrets != nil {
+		return cachedSecrets, nil
+	}
+
+	if !s.isSecurityEnabled() {
```

Comment:
Move to the top or func since not using cache when getting from insecure secrets. No need to cache them and don't want to in case user changes them from Consul.